src/main/groovy/com/palantir/python/miniconda/MinicondaPlugin.groovy

```diff
@@ -75,36 +108,39 @@ class MinicondaPlugin implements Plugin<Project> {
                 }
             }
 
-            def minicondaBootstrapVersionDir = new File(myExt.bootstrapDirectory, myExt.minicondaVersion)
-            project.task([type: Exec], "bootstrapPython") {
-                outputs.dir(minicondaBootstrapVersionDir)
-                onlyIf {
-                    !minicondaBootstrapVersionDir.exists()
-                }
+            bootstrapPython.configure {
+                inputs.property "version", myExt.minicondaVersion
+                inputs.property "directory", myExt.bootstrapDirectoryPrefix
+                outputs.dir myExt.bootstrapDirectory
+
                 if (os.contains("Windows")) {
-                    commandLine conf.singleFile, "-b", "-p", minicondaBootstrapVersionDir
+                    commandLine conf.singleFile, "-b", "-p", myExt.bootstrapDirectory
                 } else {
-                    commandLine "bash", conf.singleFile, "-b", "-p", minicondaBootstrapVersionDir
+                    commandLine "bash", conf.singleFile, "-b", "-p", myExt.bootstrapDirectory
                 }
                 doFirst {
-                    if (!myExt.bootstrapDirectory.exists()) {
-                        myExt.bootstrapDirectory.mkdir()
+                    if (!myExt.bootstrapDirectoryPrefix.exists()) {
+                        myExt.bootstrapDirectoryPrefix.mkdirs()
+                    }
+                    if (myExt.bootstrapDirectory.exists()) {
+                        project.delete myExt.bootstrapDirectory
                     }
                 }
+
+                onlyIf { !myExt.bootstrapDirectory.exists() }
             }
 
-            project.task("setupPython") {
-                dependsOn "bootstrapPython"
-                inputs.property("packages", myExt.packages)
+            setupPython.configure {
+                inputs.property "packages", myExt.packages
                 outputs.dir myExt.buildEnvironmentDirectory
+
+                executable Paths.get("$myExt.bootstrapDirectory", 'bin', 'conda')
```

Comment:
Gotcha, thanks for explaining. I have a slight personal preference for the "new File(...)" style but this is okay too.